packages/xstate-test/test/plans.test.ts

```diff
@@ -0,0 +1,149 @@
+import { createTestModel } from '../src';
+import { coversAllStates } from '../src/coverage';
+import { createTestMachine } from '../src/machine';
+import { testUtils } from '../src/testUtils';
+
+const multiPathMachine = createTestMachine({
+  initial: 'a',
+  states: {
+    a: {
+      on: {
+        EVENT: 'b'
+      }
+    },
+    b: {
+      on: {
+        EVENT: 'c'
+      }
+    },
+    c: {
+      on: {
+        EVENT: 'd',
+        EVENT_2: 'e'
+      }
+    },
+    d: {},
+    e: {}
+  }
+});
+
+describe('testModel.testPlans(...)', () => {
+  it('custom plan generators can be provided', async () => {
+    const testModel = createTestModel(
+      createTestMachine({
+        initial: 'a',
+        states: {
+          a: {
+            on: {
+              EVENT: 'b'
+            }
+          },
+          b: {}
+        }
+      })
+    );
+
+    const plans = testModel.getPlans({
+      planGenerator: (behavior, options) => {
+        const events = options.getEvents?.(behavior.initialState) ?? [];
+
+        const nextState = behavior.transition(behavior.initialState, events[0]);
+        return [
+          {
+            state: nextState,
+            paths: [
+              {
+                state: nextState,
+                steps: [
+                  {
+                    state: behavior.initialState,
+                    event: events[0]
+                  }
+                ],
+                weight: 1
+              }
+            ]
+          }
+        ];
+      }
+    });
+
+    await testUtils.testPlans(testModel, plans);
+
+    expect(testModel.getCoverage(coversAllStates())).toMatchInlineSnapshot(`
+      Array [
+        Object {
+          "criterion": Object {
+            "description": "Visits \\"(machine)\\"",
+            "predicate": [Function],
+            "skip": false,
```

Comment:
I can't find any test that would end up with `skip: true`, could you create one?

Reply:
I could have sworn I had a test that had criterion skipping; @mattpocock this is for skipping coverage w/r/t coverage testing, not for skipping tests.

We talked about _not_ testing coverage though, which makes sense. If we agree on that (and remove `.testCoverage(...)`, then yes, we can remove coverage skipping.

Reply:
@davidkpiano I agree that we should remove `.testCoverage`, I'll add a PR to remove it.

Reply:
Criterion removed